### PR TITLE
feat(company role): support optional agreements

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Support optional agreements for company role

## Why

To support optional agreements with validation

## Issue

https://github.com/eclipse-tractusx/portal-frontend-registration/issues/130

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

**Review by nidhigarg-bmw — APPROVED** [review]
lgtm

File: src/components/cax-companyRole.tsx [hunk 1]
```diff
@@ -197,33 +205,54 @@ export const CompanyRoleCax = () => {
...
+        agreements,
+      }
+
+      await updateAgreementConsents({ applicationId, data })
+        .unwrap()
+        .then(() => {
+          dispatch(addCurrentStep(currentActiveStep + 1))
+        })
+        .catch((errors) => {
+          console.log('errors', errors)
```
Comment (oyo) [inline src/components/cax-companyRole.tsx:252]: please remove console log

**Review by oyo — CHANGES_REQUESTED** [review]
guess that console.log was forgotten

File: src/components/cax-companyRole.tsx (hunk 1 above)
Comment (lavanya-bmw) [reply src/components/cax-companyRole.tsx]: Removed console

**Review by oyo — DISMISSED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**